### PR TITLE
feat: highlight the selection in the source buffer

### PR DESCRIPTION
**Pull request by folke**
I added a new option `highlight_selection`, that highlights the selection in the source buffer when in the chat window.
The highlight is only visible when the chat window is focused.

The hl group for the selection is `CopilotChatSelection` and is linked to `Visual` by default.

The option is enabled by default.

Let me know what you think!

image

### CODE REVIEW

File: lua/CopilotChat/init.lua [hunk 1]
```diff
@@ -275,6 +275,34 @@ function M.prompts(skip_system)
   return prompts_to_use
 end
 
+local selection_ns = vim.api.nvim_create_namespace('copilot-chat-selection')
+
+--- Highlights the selection in the source buffer.
+---@param opts? {clear?: boolean}
+function M.highlight_selection(opts)
```
Comment (deathbeam) [inline lua/CopilotChat/init.lua:282]: Opts can just be boolean, if highlight will be more configurable in future we usually put defaults to config and then allow config object to be passed around

**Review by deathbeam — CHANGES_REQUESTED** [review]
Otherwise looks good

**Comment by folke** [comment]
All done!

**Review by deathbeam — APPROVED** [review]
That was quick, looks good thanks!

**Comment by deathbeam** [comment]
@all-contributors add @folke for code, docs

**Comment by folke** [comment]
Thanks for merging!

**Pull request merged**